src/models/BaseLoginModel.js

```diff
@@ -58,8 +58,8 @@ function (Okta, Q, Enums) {
               err.name === Enums.AUTH_STOP_POLL_INITIATION_ERROR) {
             return;
           }
-          self.trigger('setTransactionError', err);
           self.trigger('error', self, err.xhr);
+          self.trigger('setTransactionError', err);
```

Comment:
Changing this order grants us the ability to add additional listener logic to the `transactionError` event being emitted (see [`BaseLoginController`](https://github.com/okta/okta-signin-widget/pull/594/files#diff-e7366467f1d90e02ae84e1dab0578a09)), as we know that the error has been rendered to the page.

The `setTransactionError` and `error` events are not dependent on one another, so this change is safe.

Reply:
hmm, the only soft dependency I'm seeing is when INVALID_TOKEN_EXCEPTION

before:
- navigate back to primary auth hence `model.trigger('error')` actually wont happen

after
- model.trigger('error') hence form renders error banner
- navigate back to primary auth

sounds correct? though it may not be big concern.

Reply:
After writing a few more test cases for this, it turns out that there are some cases where we're creating two banners after reordering these. I'll go back to the drawing board 😞

Reply:
why/how it renders two banner after your reordering?

Reply:
There was an extra `trigger('error')` being called in a failure handler. I'll leave a comment on my next commit with more details.

Reply:
will look at the future comment, this is an interesting topic. Looking at `RouterUtil` I see the same as Haisheng was point out, and seems that `flashError` is the special handling for it... so we might need to do something there as well.